### PR TITLE
Adds a -artifact-output-file option to save the ami_id (ref #1701)

### PR DESCRIPTION
**Pull request by pcboy**
As seen in #1701 or [parse AMI id from packer output](https://groups.google.com/forum/#!topic/packer-tool/FjwlZU2lo3g), a decent amount of people need to easily get the ami_id after generation.
Right now it's not very convenient to get (need to parse the output).
This pull request adds a -artifact-output-file parameter to the build command.
Now a call to `packer build -artifact-output-file=ami_id packer.json` will generate the ami and save the ami's id in the file ami_id.

### CODE REVIEW

**Comment by sethvargo** [comment]
This looks really good @pcboy! Do you think you could add a test for this? At least something simple for the CLI flag?

**Comment by pcboy** [comment]
@sethvargo I added some test but I'm not well versed in go so I may have done shit.

File: command/build.go [hunk 1]
```diff
@@ -34,6 +35,7 @@ func (c BuildCommand) Run(args []string) int {
 	cmdFlags.BoolVar(&cfgDebug, "debug", false, "debug mode for builds")
 	cmdFlags.BoolVar(&cfgForce, "force", false, "force a build if artifacts exist")
 	cmdFlags.BoolVar(&cfgParallel, "parallel", true, "enable/disable parallelization")
+	cmdFlags.StringVar(&artifactOutputFile, "artifact-output-file", "", "Filename where to store the created instance id")
```
Comment (jszwedko) [inline command/build.go]: Should this say `artifact` rather than `instance`?

**Comment by jszwedko** [comment]
What are the semantics of something like the `amazon-ebs' provider that could have multiple artifacts. Also what if you have multiple builders in a single file? Seems like you would only get the artifact of the last one.

**Comment by mitchellh** [comment]
I like this! I have some pedantic nitpicks but the logic is sound. I'll work on getting this in.

**Comment by mitchellh** [comment]
Actually, there is some uncertainty for me about the format that the output should be in, so I'm going to hold off for now.

**Comment by bemehow** [comment]
Can this be better served in the post-processor? Currently local-shell is ignoring aws artifacts because they have no `Files()` property but if that logic can be easily implemented there if it did not ignore with just a shell command like this: ```
    "post-processors": [
      {
        "type": "shell-local",
        "inline": ["echo"],
        "execute_command" : "echo {{.Artifact.Description}} | process_raw_Artifact.Description.py"
      }
    ]
```

AFAIR Description is where the region => ami-id mapping is getting put after the artifact is built for the AWS

**Comment by mwhooker** [comment]
Thanks for the PR. we decided to implement this with the manifest. Read more here https://www.packer.io/docs/post-processors/manifest.html

**Pull request closed without merging**